### DISCUSSION TITLE
Rendering children

### DISCUSSION THREAD

**Discussion by estebanpanelli** [react-page/react-page#978] [Q&A]
Hi all,

I have a cell pluguin for defining custom grid layouts. It was working fine in version 1.x.x.

Since updating to 2.x.x I'm having a problem to make it work because the editor wraps children inside a div tag and the parent element (who has the grid properties) has always only one child.  

There is a way of rendering children without wrapping them in a div element?

Thanks in advance.

**Comment by macrozone** [comment]
@estebanpanelli that's probably a regression from the cell-spacing change. But the surounding div should be also the only child and should stretch to its surroundings, so you should be able to style its child (the component of the CellPlugin) as you wish. Can you describe the problem you are facing?

**Reply by estebanpanelli** [reply]
Hi Marco, thanks for your answer.

As its shown in the capture, the children are always rendered inside a div that i can't control and makes it impossible to make the grid to work.

![rendererproblem](https://user-images.githubusercontent.com/22087348/116124727-0e994700-a69b-11eb-95ad-6ff2f8d64810.png)

I know it's redundant to have the 2 grids, but thats because I'm testing in both ways. Here is the cell code:

![rendererproblem2](https://user-images.githubusercontent.com/22087348/116125891-68e6d780-a69c-11eb-8774-3b48bed21491.png)

**Reply by macrozone** [reply]
i see, that was this change: https://github.com/react-page/react-page/pull/919/files#diff-6047cf581ac2c813b857adb169428015ae1632e7ac48e3651317b3a27410c8bbL85

@avoskresensky We introduced some new divs with the cell-spacing change that leads now to some troubles. 

You proposed custom Rows and Cells (https://github.com/react-page/react-page/issues/969). Do you think you can solve the cell-spacing just with this new api? Custom Row and cell wrapper could also give you the possibility to do cell-spacing different for different viewports, so i am wondering if we should undo the cell-spacing-changes (for v3) in favor of a custom Rows and Cells. What do you think?

**Reply by avoskresensky** [reply]
@macrozone , custom Cell components can be of use here. Question is, which case we should treat as more common: cell spacing or an ability to have alternative grid layouts. You can probably guess my preference ;)

**Comment by avoskresensky** [comment]
@estebanpanelli as an immediate solution you can move your grid styles to that inner div via css: assign a className to it and add a CSS rule:
```
.your-class > div {
   display: grid;
   ...
}
```

**Reply by estebanpanelli** [reply]
Thank you both for your answers. 

Please, let me express my opinion regarding this cellSpacing functionality.

I think it's an overkill to have a global config for a cell spacing when you have many other ways to achieve this using CSS or cellStyle option of the cell.

Maybe I just don't get the benefits of having this feature.
